Trees/DepthFirstSearch.js

```diff
@@ -1,76 +1,213 @@
 /*
- * Author: Surendra Kumar
+ * Author: Sundaram Kumar Jha
  * DFS Algorithm implementation in JavaScript
  * DFS Algorithm for traversing or searching graph data structures.
 */
 
-function traverseDFS (root) {
-  const stack = [root]
-  const res = []
+function dfs(tree) {
+  if (!tree) return;
+  console.log(tree.value);
+  for (const child of tree.children) {
+    dfs(child);
+  }
+}
 
-  while (stack.length) {
-    const curr = stack.pop()
-    res.push(curr.key)
+// This function takes a tree node as an argument, and uses recursion to traverse 
+// the tree in a depth-first manner. The function first logs the value of the current
+//  node and then iterates through its children, calling the function on each child. 
+// The function stops when it encounters a null node.
 
-    if (curr.right) {
-      stack.push(curr.right)
-    }
+function searchDFS(tree, target) {
+  if (tree.value === target) {
+    return true;
+  }
 
-    if (curr.left) {
-      stack.push(curr.left)
+  for (let i = 0; i < tree.children.length; i++) {
+    if (searchDFS(tree.children[i], target)) {
+      return true;
     }
   }
 
-  return res.reverse()
+  return false;
 }
 
-function searchDFS (tree, value) {
-  const stack = []
+// This function takes two arguments: tree, which is the tree to be searched, and target, 
+// which is the value you're searching for. The function uses a depth-first search algorithm to 
+// traverse the tree and look for the target value.
 
-  stack.push(tree[0])
+// The function starts by checking if the current node has the target value. 
+// If it does, it returns true to indicate that the target value has been found. 
+// If not, the function loops over the children of the current node and calls itself recursively on each child. 
+// If any of the recursive calls returns true, the function also returns true to indicate that the target value 
+// has been found. If none of the recursive calls returns true, the function returns false to indicate that the 
+// target value was not found in the tree
 
-  while (stack.length !== 0) {
-    for (let i = 0; i < stack.length; i++) {
-      const node = stack.pop()
+const tree = {
```

Comment:
This should be in a proper test case.

Reply:
can you provide some more info on that  like what kind of test you want,

it will help me to make changes then....

Reply:
See https://github.com/TheAlgorithms/TypeScript/blob/master/CONTRIBUTING.md#writing-good-tests